### ISSUE TITLE
Progressive FPS drops

### ISSUE THREAD

**Issue by akhamar** [mircearoata/SatisfactoryAdvancedLogistics#19]
I was using the mod to load balance properly my input / output. The mod works really great for that !
Sadly I've been forced to remove them all and switch back to regular spliter/merger as this mod progressively cut FPS during playtime until I reach 10FPS (from initially 140 fps)
Without the mod I've a constant FPS.

**Comment by ajhix36** [comment]
I'm having this exact same issue. Reloading save will fix, and within 5-10 mins, my FPS is completely decimated.

**Comment by akhamar** [comment]
@ajhix36 I've switched back to regular spliter a long time ago. No more problems. You have to put more splitter, but at least i'm at 100fps with a massive huge megafactory ^^

Hopefully there will be a fix.

**Comment by Alighierian** [comment]
Ditto. Just tested some of the mods I thought might be causing it, which ended up being this one. It works well enough at very limited quantities, but ramp up the use and it starts eating my CPU's capacity, and consequently my FPS whilst toasting my CPU, within mere minutes at this point.
Which is a shame, as it's pretty useful for manufacturing block setups.

**Comment by kminooie** [comment]
have the same problem as the op. the fps goes down to 5 after about 20 minutes after loading.

[gw1_03_bfps3.sav.zip](https://github.com/mircearoata/SatisfactoryAdvancedLogistics/files/6758252/gw1_03_bfps3.sav.zip)

**Comment by rjch-au** [comment]
I'm likewise having the same issue.  I've definitely isolated it to the Advanced Logistics mod - I can happily enable both the ExoSuit and SMART! mods and FPS remains steady.  If I enable Advanced Logistics _only_, I get the FPS slowdowns.  After a load, FPS is 60+, however within a couple of minutes it's down to 40 or so, but within 10 minutes it's down to a very noticeable 20.  After 20 minutes or so, the game is essentially unplayable and down to around 12-15 FPS.

This isn't a CPU or GPU bottleneck - if I look at my task manager, the CPU and GPU are close to maxed out immediately after a load, but both CPU and GPU usage slowly drops as the FPS gets worse.

**Comment by MaximusGH** [comment]
when using separators of more than 15-20 pieces, the FPS sags in the wildest way. 10-15 minutes after the start of the game. From 120 to 10-15.

**Comment by Tauschung** [comment]
Confirming issue still present as of Update 5, build 174799.

**Comment by greyishmoon** [comment]
More feedback on issue - Update 5 build 176027.
Same drop in from 60fps down to 12 over 5-10 mins. Reload back to 60 then progressive drop again. Mid-early setup with 50-60 Adjustable Splitters and 30-40 Adjustable Mergers. Adjusted graphics settings down to lowest settings (on everything) - same symptoms = 120+ fps drops to 12-15 over 10 mins. Testing showed same problem with just this mod running (no mod clash).

After seeing the known issue note on the mod DB page, went through and removed all Adjustable Mergers and problem resolved - able to maintain 60fps on max graphics settings still using Adjustable Splitters pretty well everywhere (currently 80+).

TLDR - Adjustable Splitters are working fine for me  - problems seems to solely be caused by Adjustable Mergers.

Thank you for the mod - really enjoy the functionality it provides :) I can live without mergers, but would really miss the splitters - hopefully others will see this and be able to continue using the splitters without ditching the mod entirely.

**Comment by mircearoata** [comment]
(Finally) fixed in v0.3.4

**Comment by Tauschung** [comment]
Thanks for the fix!
